homeassistant/components/proxmox/binary_sensor.py

```diff
@@ -0,0 +1,56 @@
+"""Support for binary sensors to display Proxmox VE data."""
+import homeassistant.components.proxmox as proxmox
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up Proxmox VE binary sensor."""
+    nodes = hass.data[proxmox.DATA_PROXMOX_NODES]
+    sensors = []
+
+    for node_name in nodes.keys():
+        item = nodes[node_name]
+        if 'type' not in item or item['type'] != 'node':
+            sensor = PXMXBinarySensor(hass, node_name, 'Status')
+            sensors.append(sensor)
+
+    async_add_entities(sensors)
+
+
+class PXMXBinarySensor(BinarySensorDevice):
+    """Define a binary sensor for Proxmox VE VM/Container state."""
+
+    def __init__(self, hass, node_name, sensor_name):
+        """Initialize Proxmox VE binary sensor."""
+        self._hass = hass
+        self._node_name = node_name
+        self._sensor_name = sensor_name
+        self._is_available = False
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} ({})'.format(self._sensor_name, self._node_name)
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
```

Comment:
I have the following suggestions:
- add a function for `unique_id`
- add the vmid in the `device_state_attributes`, because if the user changes the name of the entity, he can't directly see the vmid anymore
- you can also use `device_class=connectivity` to show a nice server icon